engines/freescape/freescape.h

```diff
@@ -0,0 +1,354 @@
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef FREESCAPE_H
+#define FREESCAPE_H
+
+#include "common/bitarray.h"
+#include "common/random.h"
+#include "engines/advancedDetector.h"
+#include "engines/engine.h"
+#include "graphics/palette.h"
+#include "graphics/surface.h"
+#include "graphics/tinygl/pixelbuffer.h"
+
+#include "audio/decoders/wave.h"
+#include "audio/mixer.h"
+#include "audio/softsynth/pcspk.h"
+
+#include "freescape/area.h"
+#include "freescape/gfx.h"
+#include "freescape/objects/entrance.h"
+#include "freescape/objects/geometricobject.h"
+
+namespace Freescape {
+
+class Renderer;
+
+#define FREESCAPE_DATA_BUNDLE Common::String("freescape.dat")
+
+enum CameraMovement {
+	kForwardMovement,
+	kBackwardMovement,
+	kLeftMovement,
+	kRightMovement
+};
+
+typedef Common::HashMap<uint16, Area *> AreaMap;
+typedef Common::Array<byte *> ColorMap;
+typedef Common::HashMap<uint16, int32> StateVars;
+typedef Common::HashMap<uint16, uint32> StateBits;
+
+enum {
+	kFreescapeDebugMove = 1 << 0,
+	kFreescapeDebugParser = 1 << 1,
+	kFreescapeDebugCode = 1 << 2,
+	kFreescapeDebugMedia = 1 << 4,
+};
+
+struct entrancesTableEntry {
+	int id;
+	int position[3];
+};
+
+struct soundFx {
+	int size;
+	int sampleRate;
+	byte *data;
+};
+
+class FreescapeEngine : public Engine {
+private:
+	// We need random numbers
+	Common::RandomSource *_rnd;
+
+public:
+	FreescapeEngine(OSystem *syst, const ADGameDescription *gd);
+	~FreescapeEngine();
+
+	const ADGameDescription *_gameDescription;
+	bool isDemo() const;
+
+	// Game selection
+	Common::String _variant;
+	bool isDriller() { return _targetName.hasPrefix("driller") || _targetName.hasPrefix("spacestationoblivion"); }
+	bool isDark() { return _targetName.hasPrefix("darkside"); }
+	bool isEclipse() { return _targetName.hasPrefix("totaleclipse"); }
+	bool isCastle() { return _targetName.hasPrefix("castle"); }
+	bool isAmiga() { return _targetName.hasSuffix("-amiga") || Common::matchString(_targetName.c_str(), "*-amiga-#"); }
+	bool isAtariST() { return _targetName.hasSuffix("-st") || Common::matchString(_targetName.c_str(), "*-st-#"); }
+	bool isDOS() { return !isAmiga() && !isAtariST(); }
```

Comment:
It would be better to check `_gameDescription->platform` rather than the target name.

Reply:
Fixed!